cmake/toolchain/xtools/target.cmake

```diff
@@ -10,7 +10,11 @@ set(CROSS_COMPILE_TARGET_arc         arc-zephyr-elf)
 set(CROSS_COMPILE_TARGET_x86      x86_64-zephyr-elf)
 set(CROSS_COMPILE_TARGET_sparc     sparc-zephyr-elf)
 
-set(CROSS_COMPILE_TARGET ${CROSS_COMPILE_TARGET_${ARCH}})
+if(DEFINED ENV{ZEPHYR_CROSS_COMPILE_TARGET})
+  set(CROSS_COMPILE_TARGET "$ENV{ZEPHYR_CROSS_COMPILE_TARGET}")
+else()
+  set(CROSS_COMPILE_TARGET ${CROSS_COMPILE_TARGET_${ARCH}})
+endif()
```

Comment:
This seems to be adding to this mess: https://github.com/zephyrproject-rtos/zephyr/issues/40389

Also there is no documentation of this variable, so how are users expected to know about it ?

Reply:
Naming suggestions welcome; as you say, there isn't much consistency here. I figured that using a ZEPHYR_ prefix was a bit more in line with similar usage elsewhere. However, reviewing the crosstool-ng docs, it seems like using XTOOLS_ would be a better choice

I've changed the name to match the other crosstools-ng environment variable names and also included a patch for the docs.